acceptance/acceptance_test.go

```diff
@@ -0,0 +1,205 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/cucumber/godog"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func TestFeatures(t *testing.T) {
+	suite := godog.TestSuite{
+		ScenarioInitializer:  initializeScenario,
+		TestSuiteInitializer: initializeTestSuite,
+		Options: &godog.Options{
+			Format:   "pretty",
+			Paths:    []string{"features"},
+			TestingT: t,
+		},
+	}
+
+	if suite.Run() != 0 {
+		t.Fatal("non-zero status returned, failed to run feature tests")
+	}
+}
+
+func initializeScenario(sc *godog.ScenarioContext) {
+	sc.Before(setupScenario)
+	sc.After(teardownScenario)
+
+	sc.Step(`^a source file "([^"]*)":$`, createSourceFile)
+	sc.Step(`^artifact is created for file "([^"]*)"$`, createArtifactForFile)
+	sc.Step(`^artifact is extracted for file "([^"]*)"$`, useArtifactForFile)
+	sc.Step(`^the restored file "([^"]*)" should match its source$`, restoredFileShouldMatchSource)
+}
+
+func initializeTestSuite(suite *godog.TestSuiteContext) {
+	ctx := context.Background()
+	suite.BeforeSuite(func() {
+		if err := buildContainerImage(ctx); err != nil {
+			panic(err)
+		}
+	})
+}
+
+func setupScenario(ctx context.Context, sc *godog.Scenario) (context.Context, error) {
+	tempDir, err := os.MkdirTemp("", "ec-policies-")
+	if err != nil {
+		return ctx, fmt.Errorf("setting up scenario: %w", err)
+	}
+
+	ts := testState{contextDir: tempDir}
+	if err := ts.mkdirs(); err != nil {
+		return ctx, fmt.Errorf("setting up scenario: %w", err)
+	}
+
+	return setTestState(ctx, ts), nil
+}
+
+func teardownScenario(ctx context.Context, sc *godog.Scenario, _ error) (context.Context, error) {
+	// Purposely ignore errors here to prevent a teardown error to mask a test error.
+	ts, _ := getTestState(ctx)
+	_ = ts.teardown()
+	return ctx, nil
+}
+
+func createSourceFile(ctx context.Context, fname string, content *godog.DocString) (context.Context, error) {
+	ts, err := getTestState(ctx)
+	if err != nil {
+		return ctx, fmt.Errorf("createSourceTextFile get test state: %w", err)
+	}
+
+	fpath := filepath.Join(ts.sourceDir(), fname)
+
+	f, err := os.Create(fpath)
+	if err != nil {
+		return ctx, fmt.Errorf("creating %s file: %w", fpath, err)
+	}
+	defer f.Close()
+
+	if _, err := f.WriteString(content.Content); err != nil {
+		return ctx, fmt.Errorf("writing %s file: %w", fpath, err)
+	}
+
+	return ctx, nil
```

Comment:
A bit simpler would be to use `os.WriteFile`:
```suggestion
	return ctx, os.WriteFile(fpath, []byte(content.Content), 0400)
```

Reply:
I always forget about this function! Thank you. Done.